From 9314bb25d1e8614f05cb3cae9e623edae1ca37e2 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Tue, 4 Dec 2012 17:01:38 +0100
Subject: [PORT FROM MAIN] wifi: [JB MR1 Bring Up]: revert "[REVERT ME] Disable the building process for hardware/ti/wlan initially"

BZ: 70883

revert commit 047ca567ebddb37f2e063d0c4cc0ec9356de1a3b.
and remove folder wl1271

Change-Id: Icac0954ade6e2dceb4d92adf970478c19da76d94
Orig-Change-Id: Iff30ff57b8a5206450b36569ddcc97cf6c54e1c9
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
Reviewed-on: http://android.intel.com:8080/78894
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 calibrator/Android.mk        |   28 +++++++++++++++++++++++
 wl1271/config/Android.mk.rej |   50 ------------------------------------------
 wlan_prov/Android.mk         |   44 ++++++++++++++++++++++++++++++++++++
 3 files changed, 72 insertions(+), 50 deletions(-)
 create mode 100644 calibrator/Android.mk
 delete mode 100644 wl1271/config/Android.mk.rej
 create mode 100644 wlan_prov/Android.mk

diff --git a/calibrator/Android.mk b/calibrator/Android.mk
new file mode 100644
index 0000000..8b1054a
--- /dev/null
+++ b/calibrator/Android.mk
@@ -0,0 +1,28 @@
+LOCAL_PATH:= $(call my-dir)
+
+#
+# Calibrator
+#
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES := \
+		nvs.c \
+		misc_cmds.c \
+		calibrator.c \
+		plt.c \
+		ini.c
+
+LOCAL_CFLAGS := -DCONFIG_LIBNL20
+LOCAL_LDFLAGS := -Wl,--no-gc-sections
+
+LOCAL_C_INCLUDES := \
+	$(LOCAL_PATH) \
+	external/libnl-headers
+
+LOCAL_STATIC_LIBRARIES := libnl_2
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE := calibrator
+
+LOCAL_SHARED_LIBRARIES:= libcutils
+
+include $(BUILD_EXECUTABLE)
diff --git a/wl1271/config/Android.mk.rej b/wl1271/config/Android.mk.rej
deleted file mode 100644
index f6118d5..0000000
--- a/wl1271/config/Android.mk.rej
+++ /dev/null
@@ -1,50 +0,0 @@
---- /dev/null
-+++ wl1271/config/Android.mk
-@@ -0,0 +1,47 @@
-+#
-+# Copyright (C) 2008 The Android Open Source Project
-+#
-+# Licensed under the Apache License, Version 2.0 (the "License");
-+# you may not use this file except in compliance with the License.
-+# You may obtain a copy of the License at
-+#
-+#      http://www.apache.org/licenses/LICENSE-2.0
-+#
-+# Unless required by applicable law or agreed to in writing, software
-+# distributed under the License is distributed on an "AS IS" BASIS,
-+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-+# See the License for the specific language governing permissions and
-+# limitations under the License.
-+#
-+LOCAL_PATH := $(call my-dir)
-+
-+########################
-+
-+include $(CLEAR_VARS)
-+LOCAL_MODULE := tiwlan.ini
-+LOCAL_MODULE_TAGS := user
-+LOCAL_MODULE_CLASS := ETC
-+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/wifi
-+LOCAL_SRC_FILES := $(LOCAL_MODULE)
-+include $(BUILD_PREBUILT)
-+
-+########################
-+
-+include $(CLEAR_VARS)
-+LOCAL_MODULE := dhcpcd.conf
-+LOCAL_MODULE_TAGS := user
-+LOCAL_MODULE_CLASS := ETC
-+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/dhcpcd
-+LOCAL_SRC_FILES := android_dhcpcd.conf
-+include $(BUILD_PREBUILT)
-+
-+########################
-+
-+WIFI_DRIVER_SOCKET_IFACE := tiwlan0
-+ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_6_X)
-+  include external/wpa_supplicant_6/wpa_supplicant/wpa_supplicant_conf.mk
-+else
-+  include external/wpa_supplicant/wpa_supplicant_conf.mk
-+endif
-+
-+########################
diff --git a/wlan_prov/Android.mk b/wlan_prov/Android.mk
new file mode 100644
index 0000000..3b1e09a
--- /dev/null
+++ b/wlan_prov/Android.mk
@@ -0,0 +1,44 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+ifeq ($(BUILD_WITH_CHAABI_SUPPORT),true)
+LOCAL_C_INCLUDES = \
+	$(TARGET_OUT_HEADERS)/chaabi
+endif
+
+LOCAL_SRC_FILES:= \
+	wlan_provisioning.c
+
+LOCAL_CFLAGS:=
+
+ifeq ($(BUILD_WITH_CHAABI_SUPPORT),true)
+LOCAL_CFLAGS += -DBUILD_WITH_CHAABI_SUPPORT
+LOCAL_STATIC_LIBRARIES := \
+	CC6_UMIP_ACCESS CC6_ALL_BASIC_LIB
+endif
+
+ifeq ($(TARGET_PRODUCT),mfld_gi)
+LOCAL_CFLAGS += -DSINGLE_BAND
+endif
+
+LOCAL_SHARED_LIBRARIES := \
+	libc libcutils libhardware_legacy libcrypto
+
+LOCAL_MODULE:= wlan_prov
+LOCAL_MODULE_TAGS := optional
+
+include $(BUILD_EXECUTABLE)
+
+NVS_FILE = wl1271-nvs.bin
+
+SYMLINKS := $(addprefix $(PRODUCT_OUT)/system/etc/firmware/ti-connectivity/,$(NVS_FILE))
+$(SYMLINKS): $(LOCAL_INSTALLED_MODULE) $(LOCAL_PATH)/Android.mk
+	@echo "Symlink: $@"
+	@mkdir -p $(dir $@)
+	@rm -rf $@
+	$(hide) ln -sf $(addprefix /factory/wifi/,$(NVS_FILE)) $@
+
+ALL_DEFAULT_INSTALLED_MODULES += $(SYMLINKS)
+ALL_MODULES.$(LOCAL_MODULE).INSTALLED := \
+	$(ALL_MODULES.$(LOCAL_MODULE).INSTALLED) $(SYMLINKS)
+
-- 
1.7.4.1

